# K- Nearest Neighbors

In [67]:
import pandas as pd 
import numpy as np

In [68]:
import warnings
warnings.filterwarnings("ignore")

In [69]:
df = pd.read_csv("age&income_knn.csv")
df

Age   Income
0    21      60
1    20      55
2    22      60
3    22      61
4    23      65
5    21      62
6    25      65
7    30      70
8    31      68

In [70]:
x = df.drop('Income', axis=1)
x

Age 
0    21
1    20
2    22
3    22
4    23
5    21
6    25
7    30
8    31

In [71]:
y = df['Income']
y

0    60
1    55
2    60
3    61
4    65
5    62
6    65
7    70
8    68
Name: Income, dtype: int64

In [72]:
from sklearn.neighbors import KNeighborsRegressor
knn = KNeighborsRegressor(n_neighbors=3)

In [73]:
knn.fit(x,y)

KNeighborsRegressor(n_neighbors=3)

In [74]:
knn.predict([[22]])

array([60.33333333])

In [75]:
knn.score(x, y)

0.7401315789473687

# Linear Regression

In [76]:
from sklearn.linear_model import LinearRegression
reg = LinearRegression()

In [77]:
reg.fit(x,y)

LinearRegression()

In [78]:
m=reg.coef_
m

array([1.01551724])

In [79]:
c=reg.intercept_
c

38.62931034482757

In [80]:
reg.predict([[22]])

array([60.97068966])

In [81]:
reg.score(x,y)

0.7870258620689655

In [82]:
pred = reg.predict(x)
pred

array([59.95517241, 58.93965517, 60.97068966, 60.97068966, 61.9862069 ,
       59.95517241, 64.01724138, 69.09482759, 70.11034483])

# The accuracy score of KNN & Linear Regression model are respectively 74% & 78%. In the conclusion, I can say that linear regression algorithm is more accurate than KNN algorithm for this model. 